cmd/migrate_from_qdrant.go

```diff
@@ -375,21 +374,21 @@ func getFieldType(dataType qdrant.PayloadSchemaType) *qdrant.FieldType {
 func (r *MigrateFromQdrantCmd) migrateData(ctx context.Context, sourceClient *qdrant.Client, sourceCollection string, targetClient *qdrant.Client, targetCollection string, sourcePointCount uint64) error {
 	startTime := time.Now()
 	limit := r.BatchSize
-	offset, offsetCount, err := r.getStartOffset(ctx, sourceClient, sourceCollection)
+	offset, offsetCount, err := r.getStartOffset(ctx, targetClient, sourceCollection)
 	if err != nil {
 		return fmt.Errorf("failed to get start offset: %w", err)
 	}
 
+	bar, _ := pterm.DefaultProgressbar.WithTotal(int(sourcePointCount)).Start()
 	if offset != nil {
 		pterm.Info.Printfln("Starting from offset %s (%d)", offset, offsetCount)
+		bar.Add(int(offsetCount))
```

Comment:
Makes more sense if the total always equals to the source collection count.

But the progress starts from offset in case of restarts.